### PR TITLE
[measurement-tools] Hide Measurement Tools from toolbars when view is a sheet view.

### PR DESCRIPTION
**Pull request by cortneysmithbentley**
Added some checks to the UI Items Provider so that the toolbar button doesn't show on a sheet view.

### CODE REVIEW

**Review by aruniverse — DISMISSED** [review]
shouldnt this be done at the app level? 

arent there measurement tools tht make sense in sheet views like distance or angles or whatever?

**Comment by cortneysmithbentley** [comment]
> 



> 

I thought so too but @JoeZman and @jason-crow believed it should be here, so I'll let them defend the decision. 

I don't know if those measurements types make sense for sheets.

**Comment by jason-crow** [comment]
> 

There are measurement tools that make sense for drawings but for sheets they won't work

File: packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx [hunk 1]
```diff
@@ -64,6 +67,10 @@ export class MeasureToolsUiItemsProvider implements UiItemsProvider {
       }
 
       if (toolbarOrientation === ToolbarOrientation.Vertical) {
+        // don't display measurement tools on a sheet view
+        const vp = IModelApp.viewManager.selectedView;
```
Comment (jason-crow) [inline packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx:71]: Don't add handling here. The tool can be added regardless of the current view its just the tool will hide itself if the current view is a sheet, which could change without this code here rerunning

Comment (cortneysmithbentley) [reply packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx:71]: where would I add it then?

File: packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx [hunk 2]
```diff
@@ -64,6 +67,10 @@ export class MeasureToolsUiItemsProvider implements UiItemsProvider {
...
+        if (vp && ViewHelper.isSheetView(vp))
+          return [];
         return [
           ToolbarItemUtilities.createGroupButton(
```
Comment (jason-crow) [inline packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx:75]: the last parameter for createGroupButton is overrides, here you could define a isHidden value that you set with a ConditionalBooleanValue that syncs with the ViewStateChange and checks if the view is a sheet

File: packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx [hunk 3]
```diff
@@ -80,10 +87,14 @@ export class MeasureToolsUiItemsProvider implements UiItemsProvider {
 
       if (tools.length > 0 && toolbarOrientation === ToolbarOrientation.Horizontal) {
         const isHidden = new ConditionalBooleanValue(
-          () => !MeasurementUIEvents.isClearMeasurementButtonVisible,
+          () => {
+            const vp = IModelApp.viewManager.selectedView;
+            return (vp ? !MeasurementUIEvents.isClearMeasurementButtonVisible || ViewHelper.isSheetView(vp) : !MeasurementUIEvents.isClearMeasurementButtonVisible);
```
Comment (jason-crow) [inline packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx:92]: why not
!MeasurementUIEvents.isClearMeasurementButtonVisible && ViewHelper.isSheetView(vp)

**Review by jason-crow — APPROVED** [review]

**Review by aruniverse — APPROVED** [review]

**Pull request merged**